components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -124,389 +130,410 @@
+            PrivilegedCarbonContext.endTenantFlow();
         }
-        return null;
     }
 
-    @GET
+
+    @POST
     @Path("/")
     @Consumes("application/x-www-form-urlencoded")
     @Produces("text/html")
-    public Response authorize(@Context HttpServletRequest request, @Context HttpServletResponse response)
-            throws URISyntaxException {
+    public Response authorizePost(@Context HttpServletRequest request, @Context HttpServletResponse response, MultivaluedMap paramMap)
+            throws URISyntaxException, InvalidRequestParentException {
 
-        // Setting super-tenant carbon context
+        // Validate repeated parameters
+        if (!EndpointUtil.validateParams(request, paramMap)) {
+            return Response.status(HttpServletResponse.SC_BAD_REQUEST).location(new URI(
+                    EndpointUtil.getErrorPageURL(OAuth2ErrorCodes.INVALID_REQUEST,
+                            "Invalid authorization request with repeated parameters", null))).build();
+        }
+        HttpServletRequestWrapper httpRequest = new OAuthRequestWrapper(request, paramMap);
+        return authorize(httpRequest, response);
+    }
+
+    private Response handleInvalidRequest() throws URISyntaxException {
+        if (log.isDebugEnabled()) {
+            log.debug("Invalid authorization request");
+        }
+
+        return Response.status(HttpServletResponse.SC_FOUND).location(new URI(EndpointUtil.getErrorPageURL
+                (OAuth2ErrorCodes.INVALID_REQUEST, "Invalid authorization request", null))).build();
+    }
+
+    private void handleRetainCache(OAuthMessage oAuthMessage) {
+        String sessionDataKeyFromConsent = oAuthMessage.getRequest().getParameter(OAuthConstants.SESSION_DATA_KEY_CONSENT);
+        if (sessionDataKeyFromConsent != null) {
+            /*
+             * TODO Cache retaining is a temporary fix. Remove after Google fixes
+             * http://code.google.com/p/gdata-issues/issues/detail?id=6628
+             */
+            String retainCache = System.getProperty("retainCache");
+
+            if (retainCache == null) {
+                clearCacheEntry(sessionDataKeyFromConsent);
+            }
+        }
+    }
+
+    private boolean isConsentResponseFromUser(OAuthMessage oAuthMessage) {
+        return USER_CONSENT_RESPONSE.equals(oAuthMessage.getRequestType());
+    }
+
+    private boolean isAuthenticationResponseFromFramework(OAuthMessage oAuthMessage) {
+        return AUTHENTICATION_RESPONSE.equals(oAuthMessage.getRequestType());
+    }
+
+    private boolean isInitialRequestFromClient(OAuthMessage oAuthMessage) {
+        return INITIAL_REQUEST.equals(oAuthMessage.getRequestType());
+    }
+
+    private boolean isPassthroughToFramework(OAuthMessage oAuthMessage) {
+        return PASSTHROUGH_TO_COMMONAUTH.equals(oAuthMessage.getRequestType());
+    }
+
+    private OAuthMessage buildOAuthMessage(@Context HttpServletRequest request, @Context HttpServletResponse response)
```

Comment:
Do we need the @Context annotations here?

Reply:
removed